streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -1104,11 +1104,12 @@ private void initializeTopology() {
         // initialize the task by initializing all its processor nodes in the topology
         log.trace("Initializing processor nodes of the topology");
         for (final ProcessorNode<?, ?, ?, ?> node : topology.processors()) {
-            processorContext.setCurrentNode(node);
+            final InitProcessorRecordContext initContext = new InitProcessorRecordContext(time);
+            updateProcessorContext(node, time.milliseconds(), initContext);
```

Comment:
Looking into `updateProcessorContext()` it does set "system time" to a fixed value. We did discuss this originally, if it make sense to have a static ts, or a moving one, and though we want a moving one.

But if system time is not moving either, I am wondering if we really need it, or should fall back to just getting system time once, and set on the context. 🤔 

Or would we also need to update `ProcessorRecordContext` to return a "dynamic" system time? We did opt to return a static one, as it is not for free to call `System.currentTimeMillis()` in general, so we want to cache the result and reuse to avoid unnecessary overhead. If this was "good enough" so far, it might also be "good enough" during initialization, too?

Thoughts? (Sorry for the back and forth...)